src/main/scala/cromwell/engine/SingleWorkflowRunnerActor.scala

```diff
@@ -0,0 +1,77 @@
+package cromwell.engine
+
+import akka.actor.FSM.{Transition, CurrentState}
+import akka.actor.{ActorRef, Props, Actor}
+import akka.event.Logging
+import akka.pattern.ask
+import akka.util.Timeout
+import cromwell.binding
+import cromwell.binding.WdlSource
+import cromwell.engine.WorkflowManagerActor._
+import spray.json._
+import scala.concurrent.duration._
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.language.postfixOps
+import scala.util.{Success, Failure}
+
+object SingleWorkflowRunnerActor {
+  def props(wdl: WdlSource, inputs: binding.WorkflowRawInputs, workflowManager: ActorRef) = {
+    Props(new SingleWorkflowRunnerActor(wdl, inputs, workflowManager))
+  }
+}
+
+case class SingleWorkflowRunnerActor(wdl: WdlSource,
+                                     inputs: binding.WorkflowRawInputs,
+                                     workflowManager: ActorRef) extends Actor {
+  private val log = Logging(context.system, this)
+  implicit val timeout = Timeout(5 seconds)
+  // Note that id isn't used until *after* the submitWorkflow Future is complete
+  private var id: WorkflowId = _
+
+  override def preStart(): Unit = {
+    val eventualId = workflowManager.ask(SubmitWorkflow(wdl, inputs)).mapTo[WorkflowId]
+    eventualId onComplete {
+      case Success(x) => subscribeToWorkflow(x)
+      case Failure(e) =>
+        log.error(e.getMessage)
+        terminate()
+    }
+  }
+
+  def receive = {
+    case CurrentState(_, state: WorkflowState) if isTerminalState(state) => handleTermination(state)
+    case Transition(_, _, state: WorkflowState) if isTerminalState(state) => handleTermination(state)
+    case _ => // No-op
```

Comment:
I  can do that. This case goes beyond "shouldn't happen" into the "I don't even think it *can* happen", but one never knows.